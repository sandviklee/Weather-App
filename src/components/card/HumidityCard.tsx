import styles from "./HumidityCard.module.css";

interface CardProps {
  humidity: number;
}

/**
 * Returns a status for given humidity.
 * @param humidity 
 * @returns status string
 */
const convertFromRelativeHumidityToStatusString = (humidity: number) => {
  if (humidity < 30) {
    return "Lav";
  } else if (humidity >= 30 && humidity <= 60) {
    return "Normal";
  } else {
    return "HÃ¸y";
  }
};

/**
 * Returns a color for given humidity.
 * @param humidity 
 * @returns a hex color string
 */
const convertFromRelativeHumidityToColorStatus = (humidity: number) => {
  if (humidity < 30) {
    return "#FFFA8D";
  } else if (humidity >= 30 && humidity <= 60) {
    return "#B2FF8D";
  } else {
    return "#FF8D8D";
  }
};

const HumidityCard = ({ humidity }: CardProps) => {
  return (
    <div className={styles.container}>
      <p className={styles.title}>Luftfuktighet</p>
      <div className={styles.humidity_container}>
        <div className={styles.status_container}>
          <div
            className={styles.circle}
            style={{
              backgroundColor:
                convertFromRelativeHumidityToColorStatus(humidity),
            }}
          ></div>
          <p>{convertFromRelativeHumidityToStatusString(humidity)}</p>
        </div>
        <p className={styles.humidity}>{humidity}%</p>
      </div>
    </div>
  );
};

export default HumidityCard;
