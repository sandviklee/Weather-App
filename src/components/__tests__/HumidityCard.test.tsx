import { render, screen } from '@testing-library/react';
import HumidityCard from '../card/HumidityCard';

describe('HumidityCard component', () => {
    test('renders correctly with low humidity', () => {
        render(<HumidityCard humidity={20} />);
        
        const humidityStatus = screen.getByText('Lav');
        expect(humidityStatus).toBeInTheDocument();
        const humidityValue = screen.getByText('20%');
        expect(humidityValue).toBeInTheDocument();
    });

    test('renders correctly with normal humidity', () => {
        render(<HumidityCard humidity={50} />);
        
        const humidityStatus = screen.getByText('Normal');
        expect(humidityStatus).toBeInTheDocument();
        const humidityValue = screen.getByText('50%');
        expect(humidityValue).toBeInTheDocument();
    });

    test('renders correctly with high humidity', () => {
        render(<HumidityCard humidity={70} />);
        
        const humidityStatus = screen.getByText('HÃ¸y');
        expect(humidityStatus).toBeInTheDocument();
        const humidityValue = screen.getByText('70%');
        expect(humidityValue).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<HumidityCard humidity={20} />);
        expect(asFragment()).toMatchSnapshot();
    });
});