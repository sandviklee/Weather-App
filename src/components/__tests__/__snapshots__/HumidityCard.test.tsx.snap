// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HumidityCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_7ae00b"
  >
    <p
      class="_title_7ae00b"
    >
      Luftfuktighet
    </p>
    <div
      class="_humidity_container_7ae00b"
    >
      <div
        class="_status_container_7ae00b"
      >
        <div
          class="_circle_7ae00b"
          style="background-color: rgb(255, 250, 141);"
        />
        <p>
          Lav
        </p>
      </div>
      <p
        class="_humidity_7ae00b"
      >
        20%
      </p>
    </div>
  </div>
</DocumentFragment>
`;
